README.md

```diff
@@ -1,51 +1,96 @@
+
+### Easy Provisioning ###
+Setup and configuration can be done right in NGINX configuration files.
+
+### Fully Dynamic Variable-Based Sampling ###
+The module provides the ability to trace a particular session by cookie/token. [NGINX Plus](https://www.nginx.com/products/nginx/), available as part of a [commercial subscription](https://www.nginx.com/products/), enables dynamic module control via the [NGINX Plus API](http://nginx.org/en/docs/http/ngx_http_api_module.html) and [key-value store](http://nginx.org/en/docs/http/ngx_http_keyval_module.html) modules.
 
 ## Building
+Follow these steps to build the `ngx_otel_module` dynamic module on Ubuntu or Debian based systems:
 
-Install build tools and dependencies:
+Install build tools and dependencies.
 ```bash
-  $ sudo apt install cmake build-essential libssl-dev zlib1g-dev libpcre3-dev
-  $ sudo apt install pkg-config libc-ares-dev libre2-dev # for gRPC
+sudo apt install cmake build-essential libssl-dev zlib1g-dev libpcre3-dev
+sudo apt install pkg-config libc-ares-dev libre2-dev # for gRPC
 ```
 
-Configure Nginx:
+For the next step, you will need the `configure` script that is packaged with the NGINX source code. There are several methods for obtaining NGINX sources. You may choose to [download](http://hg.nginx.org/nginx/archive/tip.tar.gz) them or clone them directly from the NGINX Github repository.
 ```bash
-  $ ./configure --with-compat
+git clone https://github.com/nginx/nginx.git
 ```
 
-Configure and build Nginx OTel module:
+Configure NGINX to generate files necessary for dynamic module compilation. These files will be placed into the `nginx/objs` directory. **Important:** If you did not obtain NGINX source code via the clone method in the previous step, you will need to adjust paths in the following commands to conform to your specific directory structure.
 ```bash
-  $ mkdir build
-  $ cd build
-  $ cmake -DNGX_OTEL_NGINX_BUILD_DIR=/path/to/configured/nginx/objs ..
-  $ make
+cd nginx
+auto/configure --with-compat
 ```
 
-## Getting Started
+Exit the NGINX directory and clone the `ngx_otel_module` repository.
+```bash
+cd ..
+git clone https://github.com/nginxinc/nginx-otel.git
+```
 
-### Simple Tracing
+Configure and build the NGINX OTel module.
+
+**Important**: replace the path in the `cmake` command with the path to the `nginx/objs` directory from above.
+```bash
+cd nginx-otel
+mkdir build
+cd build
+cmake -DNGX_OTEL_NGINX_BUILD_DIR=/path/to/configured/nginx/objs ..
+make
+```
+
+Compilation will produce a binary named `ngx_otel_module.so`.
+
+## Installation
+***Important:*** The built `ngx_otel_module.so` dynamic module binary will ONLY be compatible with the same version of NGINX source code that was used to build it. To guarantee proper operation, you will need to build and install NGINX from sources obtained in previous steps.
```

Comment:
Maybe you can mention this in the section where `git clone` is used. and duplicate that you need to switch to the branch of nginx that is you are planning to be using?
also worth mentioning is that you need to do it on the same OS that you intend to use (the module is built on Mac OS and used on Linux, I know it is stupid, right?)

Reply:
Yes, good point on both accounts. I've added some guidelines to the NGINX source cloning step and added that the same OS needs to be used in both places.